src/general/IFilter.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+
+public interface IFilter
```

Comment:
Why interface instead of abstract class?

Reply:
Good question. I basically wanted to hide the the type of the Filter. I spontaneously went for interface as this is what has been used for e.g. organelles, but maybe it could be better with an abstract class indeed? Haven't given it much tought.

Reply:
Out of curiosity, what "organelle interface" do you mean? Only thing I can think of is `IPositionedOrganelle` where the interface is mandatory as a Godot.Spatial derived type is needed in one place and in another we want a pure data class that has nothing to do with Godot.

Reply:
It was actually `IOrganelleComponent`, along with `IOrganelleComponentFactory` (although on my part I currently have the factory defined within the interface, but I'm open to create a new file for readability)